include/revng/Yield/Support/GraphLayout/Graphs.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <unordered_map>
+
+#include "revng/ADT/GenericGraph.h"
+#include "revng/Yield/Support/GraphLayout/Traits.h"
+
+namespace yield::layout {
+
+namespace detail {
+
+template<typename DataType>
+struct InputNode : public DataType {
+  Size Size = { 0, 0 };
+
+  using DataType::DataType;
+  InputNode(const DataType &Another) : DataType(Another){};
+  InputNode(DataType &&Another) : DataType(Another){};
+};
+
+template<typename DataType>
+struct OutputNode : public DataType {
```

Comment:
Shouldn't this inherit from `InputNode`? In a sense the `OutputNode` is just the same as the input, but with an assigned "placement" i.e. `Center`.

Reply:
Or do you see reasons not to do this?

Reply:
Now, that you have the context about not wanting the `InputNode` and `OutputNode` to be related from the user's point of view, are you also on the same page that they shouldn't be implicitly convertible, only using a separate method hidden away somewhere?